### PR TITLE
Add refinement critereon based on boundary indicator

### PR DESCRIPTION
**Pull request by ian-r-rose**
Similar to the topography refinement strategy, but the user gives a list of boundary indicators for refinement.  Sometimes the user may want refinement at the boundaries, even if there are no strong gradients there to trip other refinement criteria.

### CODE REVIEW

**Comment by gassmoeller** [comment]
Please reorder the functions in boundary.cc and boundary.h:

execute()
declare_parameters()
parse_parameters()

I just like to know that I need to look at the end for the parameter functions. 
Also, currently there is no error checking for cases like `set Boundary refinement indicators = top, left, top`, but I guess it is ok in this case, because there is no contradiction.

Otherwise this looks good to me. :+1:

File: doc/modules/changes.h [hunk 1]
```diff
@@ -7,6 +7,11 @@
  *
  * <ol>
  *
+ * <li> New: There is now a mesh refinement criterion based on
+ * refining near a user-specified list of boundary indicators.
+ * <br>
```
Comment (bangerth) [inline doc/modules/changes.h]: Can you reference the plugin by name here, so people don't have to go search for it throughout the manual?

File: source/mesh_refinement/boundary.cc [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+              {
+                const types::boundary_id boundary_indicator
+#if DEAL_II_VERSION_GTE(8,3,0)
+                  = cell->face(face_no)->boundary_id();
+#else
+                  = cell->face(face_no)->boundary_indicator();
+#endif
+                if ( boundary_refinement_indicators.find(boundary_indicator) !=
+                     boundary_refinement_indicators.end() )
+                  indicators(i) = 1.0;
```
Comment (bangerth) [inline source/mesh_refinement/boundary.cc]: if you found a cell of interest, you may as well `break` out of the loop over the faces.

**Comment by gassmoeller** [comment]
@ian-r-rose can you please rebase to current master? #330 interfered with the changes.h. After that this is ready to merge.

**Comment by ian-r-rose** [comment]
Done.

**Comment by tjhei** [comment]
Ian, can you please move your blob in changes.h one down so they are ordered by date?

**Comment by tjhei** [comment]
Would you mind rebasing to master again so we can see if the tests are passing please? (you rebased to the bad commit from yesterday).

**Comment by ian-r-rose** [comment]
How about now :) ?

**Comment by tjhei** [comment]
Thanks! Will wait for the tester to give the okay. :-)

**Pull request merged**